### PR TITLE
Remove PortMixer code

### LINKED ISSUE #517: Recording volume slider stays locked until 'Record' is clicked
### Guidelines

- [X] I have read the guidelines.

### Version/Commit hash

MSVC 19.29.30040.00

### Describe the bug.

Recording volume slider stays locked until 'Record' is clicked. 
It happens even when the meter is actively showing input volume. 
It doesn't matter if a track has been created or not.

RecordingVolumeSlider

### Expected behavior

For the input meter and input volume slider to work in unison.

### OS

Windows 10

### Additional context

_No response_

### This issue is not a duplicate

- [X] I have confirmed this issue isn't a duplicate.

### PR DESCRIPTION
**Pull request by akleja**
Resolves: https://github.com/tenacityteam/tenacity/issues/517

Since PortMixer isn't supported by upstream PortAudio it makes sense to remove it from the codebase alongside with everything that depends on it.

Maintaining functionality that isn't possible to build will result in broken legacy code and will hinder development of new solutions.

Removes:
* PortMixer + Mixer Toolbar
  It's better to set levels in audio interface instead anyway, plus it didn't work reliably with a lot of hardware.
* Experimental automated input level adjustment
  This was a bad idea from the start. Adjusting recording levels by changing hardware levels doesn't make sense to me. Set correct levels and use a compressor if needed.

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

File: src/AudioIO.cpp [hunk 1]
```diff
@@ -1057,16 +1045,6 @@ AudioIO::AudioIO()
...
    mThread->Create();
 
-#if defined(USE_PORTMIXER)
-   mPortMixer = NULL;
-   mPreviousHWPlaythrough = -1.0;
-   HandleDeviceChange();
-#else
-   mEmulateMixerOutputVol = true;
-   mMixerOutputVol = 1.0;
-   mInputMixerWorks = false;
```
Comment (Be-ing) [inline src/AudioIO.cpp:1067]: Doesn't this `else` branch need to stay but execute unconditionally?

Comment (Be-ing) [reply src/AudioIO.cpp:1067]: Oh nevermind I see all this was for PortMixer anyway.

**Comment by Be-ing** [comment]
The code looks good from a quick look but I haven't tested it yet.

> +4 −1,546

That's what I like to see! I wonder what other bad ideas we can remove.

**Comment by akleja** [comment]
> That's what I like to see! I wonder what other bad ideas we can remove. 

Seems like a lot of my contributions so far consists of removal of code

**Review by n0toose — DISMISSED** [review]
Works on my machine, not sure if any other PortMixer-related code has been left out elsewhere or if this is going to cause issues later down the line.

**Comment by akleja** [comment]
I've done my best to check for all PortMixer-code, but I could of course have missed some. But if everything is working with these removals (which it seems, though more reviews is a good idea)  if then there's something left, it has to be dead code. While not ideal, it at least shouldn't create any problems.

As with all changes it will most likely lead to merging conflicts. Also if PortMixer support is wanted again one day it would have to be reverted.

Another option I considered was to just put ifdefs around the parts not already inside ifdefs, like the toolbar and some other stuff. One downside to that approach is that it would signal that PortMixer still is an option. Having compile time options that doesn't build isn't the best idea in my mind. Another downside is that the code would probably lag behind and potentially get more and more broken as the rest of the codebase changes if we can't test it.
Yet another option would be to put everything behind ifdefs while also maintain a branch for testing of PortMixer with the old build system. But that doesn't seem all that reasonable.

**Comment by Be-ing** [comment]
> having compile time options that doesn't build isn't the best idea in my mind.

I agree. I would support removing more compile time options that aren't tested by CI.

**Comment by Be-ing** [comment]
>  Yet another option would be to put everything behind ifdefs while also maintain a branch for testing of PortMixer with the old build system. But that doesn't seem all that reasonable.

I agree that's a huge maintenance burden that would not be worth it.

**Comment by akleja** [comment]
> I agree. I would support removing more compile time options that aren't tested by CI. 

Yeah I agree, unless if there's something useful worth keep working on, but then it would be better to convert it to a branch or PR imo.
There are also those option that are always enabled, like the "new" theme engine for one. Should probably remove all the obsolete stuff for when it's disabled.

**Pull request merged**